src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Telemetry/DbConnectionFactoryTelemetry.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Data.ProviderBase;
+
+namespace Microsoft.Data.SqlClient.Telemetry
+{
+    // Stores telemetry for a single DbConnectionPoolGroup. Maintained by the connection pool group itself
+    internal sealed class DbConnectionPoolTelemetry
+    {
+        private readonly DbConnectionPoolGroupOptions _options;
+        private long _poolCount;
+
+        public long TotalMaxPoolSize => _poolCount * (long)_options.MaxPoolSize;
```

Comment:
since the options can come from user input is it worth guarding against overflow here and below?

Reply:
I don't think one could be triggered. MaxPoolSize ultimately comes via SqlConnectionString.MaxPoolSize, which is defined as an int and has protection against negative values.

An overflow would be triggered if there were `uint.MaxValue` connection pools in the connection pool group with a MaxPoolSize of `int.MaxValue`, but the list of connection pools is stored as a ConcurrentDictionary in DbConnectionPoolGroup (which is limited to storing `int.MaxValue` entries.)